.github/workflows/DetectCDKBootstrapVersionChanges.yml

```diff
@@ -1,49 +1,115 @@
-      id: currentBootstrapVersion
+        sudo wget -qO /usr/local/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
+        sudo chmod +x /usr/local/bin/yq
+
+        yq eval '.Resources.StagingBucket.UpdateReplacePolicy = "Delete"' -i newTemplate.yml
+        yq eval '.Resources.StagingBucket.DeletionPolicy = "Delete"' -i newTemplate.yml
+
+    - name: Check for version changes
+      id: check_version
       run: |
-        echo "current-version=$(yq '.Resources.CdkBootstrapVersion.Properties.Value' 'src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml')" >> $GITHUB_OUTPUT
-    - name: Fail If CDK Bootstrap Template Changes Detected
-      if: steps.currentBootstrapVersion.outputs.current-version != steps.latestBootstrapVersion.outputs.latest-version
+        OLD_VERSION=$(yq eval '.Resources.CdkBootstrapVersion.Properties.Value' src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml)
+        NEW_VERSION=$(yq eval '.Resources.CdkBootstrapVersion.Properties.Value' temp_cdk/newTemplate.yml)
+        
+        if [ "$OLD_VERSION" != "$NEW_VERSION" ]; then
+          echo "Version changed from $OLD_VERSION to $NEW_VERSION"
+          echo "version_changed=true" >> $GITHUB_OUTPUT
+          echo "new_version=$NEW_VERSION" >> $GITHUB_OUTPUT
+        else
+          echo "No version change detected"
+          echo "version_changed=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Update CDK Bootstrap Template
+      if: steps.check_version.outputs.version_changed == 'true'
+      run: |
+        cp temp_cdk/newTemplate.yml src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml
+
+    - name: Generate change file
+      if: steps.check_version.outputs.version_changed == 'true'
+      env:
+        NEW_VERSION: ${{ steps.check_version.outputs.new_version }}
+      run: |
+        dotnet tool install -g autover
+        autover change --project-name "AWS.Deploy.CLI" -m "Update CDK Bootstrap template to version $NEW_VERSION"
+
+    - name: Create Pull Request
+      if: steps.check_version.outputs.version_changed == 'true'
+      uses: peter-evans/create-pull-request@v5
+      with:
+        commit-message: 'chore: update CDK bootstrap template to version ${{ steps.check_version.outputs.new_version }}'
+        title: 'Update CDK Bootstrap Template to Version ${{ steps.check_version.outputs.new_version }}'
+        body: |
+          This PR automatically updates the CDK bootstrap template to version ${{ steps.check_version.outputs.new_version }}.
+          - Updates template to latest version
+          - Maintains DeletePolicy and UpdateReplacePolicy as "Delete" for StagingBucket
+          - Adds change file for version tracking
+        branch: update-cdk-bootstrap-template
+        base: dev
+        delete-branch: true
+        token: ${{ env.AWS_SECRET_TOKEN }}
+        add-paths: |
+          src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml
+          .autover/
+
+    - name: Cleanup
```

Comment:
nit: this step is unnecessary. The container gets deleted when the workflow is done.